requirements/base.txt

```diff
@@ -15,3 +15,4 @@ requests==2.23.0
 serverlessrepo==0.1.10
 aws_lambda_builders==1.1.0
 tomlkit==0.7.0
+dataclasses~=0.8; python_version<"3.7"
```

Comment:
Is there no other reasonable way than introducing a new dependency and using dataclasses? It seems like a reasonable change, but would be nice for the first PR to only change what's absolutely necessary.

Reply:
I agree, but without getting ride of that Dict, mypy cannot infer types in quite some places..

Reply:
I was aiming for NamedTuple at first, and then found they need to be mutable

Reply:
`dataclasses` is a pretty reputable and simple portback, I think it is pretty safe to use https://github.com/ericvsmith/dataclasses

Reply:
Yeah fair enough if not using it would be a significant detriment. I'm assuming without this mypy would complain a lot?

(Also if we merge this we should let the team know they're free to use dataclasses!)

Reply:
You know what, I changed my mind! I think it is a great suggestion to make this PR easier to go through! I made another branch with these changes and will file a PR immediately after this one is merged 😋

Reply:
They are moved here: https://github.com/aahung/aws-sam-cli/commit/b5e2b1bd34b7011cdd93192e1279e3675d77231f